src/DeserializerFactory.php

```diff
@@ -45,7 +56,7 @@ public function __construct( Deserializer $dataValueDeserializer, EntityIdParser
 	public function newEntityDeserializer() {
 		return new EntityDeserializer(
 			$this->legacyFactory->newEntityDeserializer(),
-			$this->currentFactory->newEntityDeserializer()
+			$this->currentEntityDeserializer ?: $this->currentFactory->newEntityDeserializer()
```

Comment:
I think it'd be good to describe this behaviour for the constructor parameter. Else it's not clear that the deserializer you provide gets used instead of some default.